src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/ClientMethodsAnalyzer.cs

```diff
@@ -25,8 +25,17 @@ private static void CheckClientMethod(ISymbolAnalysisContext context, IMethodSym
         {
             static bool SupportsCancellationsParameter(IParameterSymbol parameterSymbol)
             {
-                return (parameterSymbol.Name == "cancellationToken" && parameterSymbol.Type.Name == "CancellationToken") ||
-                       (parameterSymbol.Name == "context" && parameterSymbol.Type.Name == "RequestContext");
+                return IsCancellationToken(parameterSymbol) || IsRequestContext(parameterSymbol);
```

Comment:
I am not sure we exactly want a blind OR here.  If there exists a `RequestContent` then for sure it should only have a `RequestContext` and never a `CancellationToken`.

If the response is ever `Response<T>` then we shoud only have `CancellationToken` and never `RequestContext`

Not sure if there are any other fine grained checks we can do here to make this more accurate and remove edge cases.